Packs/FireEyeETP/Integrations/FireEyeETPEventCollector/FireEyeETPEventCollector.py

```diff
@@ -65,17 +67,95 @@ def __init__(
         base_url: str,
         verify_certificate: bool,
         proxy: bool,
-        api_key: str,
-        outbound_traffic: bool,
-        hide_sensitive: bool,
+        client_id: str = "",
+        client_secret: str = "",
+        scope: str = "",
+        api_key: str = "",
+        outbound_traffic: bool = False,
+        hide_sensitive: bool = False,
     ) -> None:
         super().__init__(base_url, verify_certificate, proxy)
-        self._headers = {
-            "x-fireeye-api-key": api_key,
-            "Content-Type": "application/json",
-        }
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.scope = scope
+        self.api_key = api_key
         self.outbound_traffic = outbound_traffic
         self.hide_sensitive = hide_sensitive
+        self.access_token = ""
+
+        # Set up headers based on authentication method
+        self._headers = {"Content-Type": "application/json"}
+
+        if self.client_id and self.client_secret:
+            demisto.debug(f"{LOG_LINE} Using OAuth2 authentication (Client ID/Secret)")
+            self.access_token = self._get_valid_oauth_token()
+            self._headers["Authorization"] = f"Bearer {self.access_token}"
+        elif self.api_key:
+            demisto.debug(f"{LOG_LINE} Using API Key authentication")
+            self._headers["x-fireeye-api-key"] = self.api_key
+        else:
+            raise ValueError("Either Client ID/Secret (OAuth2) or API Key (legacy) must be provided for authentication")
```

Comment:
Lets split to a different function such "get_auth_headers"

Reply:
Done